packages/protocol/contracts/stability/Broker.sol

```diff
@@ -0,0 +1,229 @@
+pragma solidity ^0.5.13;
+pragma experimental ABIEncoderV2;
+
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { IExchangeManager } from "./interfaces/IExchangeManager.sol";
+import { IBroker } from "./interfaces/IBroker.sol";
+import { IBrokerAdmin } from "./interfaces/IBrokerAdmin.sol";
+import { IReserve } from "./interfaces/IReserve.sol";
+import { Initializable } from "../common/Initializable.sol";
+
+/**
+ * @title Broker
+ * @notice The broker executes swaps and keeps track of spending limits per pair
+ */
+contract Broker is IBroker, IBrokerAdmin, Initializable, Ownable {
+  /* ==================== State Variables ==================== */
+
+  address[] exchangeManagers;
+  mapping(address => bool) isExchangeManager;
+
+  // Address of the reserve.
+  IReserve public reserve;
+
+  /* ==================== Constructor ==================== */
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialization.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice Allows the contract to be upgradable via the proxy.
+   * @param _exchangeManagers The addresses of the ExchangeManager contracts.
+   * @param _reserve The address of the Reserve contract.
+   */
+  function initilize(address[] calldata _exchangeManagers, address _reserve) external initializer {
+    _transferOwnership(msg.sender);
+    for (uint256 i = 0; i < _exchangeManagers.length; i++) {
+      addExchangeManager(_exchangeManagers[i]);
+    }
+    setReserve(_reserve);
+  }
+
+  /* ==================== Mutative Functions ==================== */
+
+  /**
+   * @notice Add exchange manager
+   * @param exchangeManager The address of the exchange manager to add
+   * @return index The index where it was inserted
+   */
+  function addExchangeManager(address exchangeManager) public onlyOwner returns (uint256 index) {
+    require(!checkIsExchangeManager(exchangeManager), "ExchangeManager already exists in the list");
```

Comment:
This and all following `checkIsExchangeManager(exchangeManager)` should be `isExchangeManager[exchangeManager]`.